src/Utils.ts

```diff
@@ -0,0 +1,103 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import Constants from "./Constants";
+import { PathUtils } from "./PathUtils";
+import MetaProperty from "./MetaProperty";
+import { ModelManager } from "./ModelManager";
+
+export class Utils {
+    /**
+     * Returns all the tags for requested state
+     *
+     * @returns {String} all the tags for requested state
+     */
+    public static getTagsForState(state : string) : string {
+        let tags = '';
+        if (state === Constants.AUTHORING) {
+            const clientLibs = this.generateClientLibsUrl();
+            clientLibs.forEach(clientlib => {
+                if (clientlib.endsWith('.js')) {
+                    tags = tags.concat(this.generateElementString('script', 'src', clientlib));
+                }
+                else if (clientlib.endsWith('.css')) {
+                    tags = tags.concat(this.generateElementString('stylesheet', 'href', clientlib));
+                }
+            });
+        }
+        return tags;
+    }
+
+    /**
+     * Returns string value of all the concatenated tags
+     *
+     * @returns {String} concatenated tags
+     */
+    private static generateElementString(tagType : string, attr : string, attrValue : string) : string {
+        let tag = '';
+        if (tagType === 'script') {
+            tag = `<script ${attr}='${attrValue}'></script>`;
+        }
+        else if (tagType === 'stylesheet') {
+            tag = `<link ${attr}='${attrValue}'/>`;
+        }
+        return tag;
+    }
+
+    /**
+     * Returns if requested state is active
+     *
+     * @returns {boolean} the result of the check of the state
+     */
+    public static isState(state : string) : boolean {
+        if (state === Constants.AUTHORING) {
+            const viaMetaProperty = PathUtils.getMetaPropertyValue(MetaProperty.WCM_MODE) === Constants.AEM_MODE_EDIT;
+            const viaQueryParam = PathUtils.isBrowser() && (Utils.getEditParam() === Constants.AEM_MODE_EDIT);
+
+            return viaMetaProperty || viaQueryParam;
+        }
+        return false;
+    }
+
+    /**
+     * Returns query parameter value for aem mode.
+     *
+     * @returns {String | null} the result of the check of the Editor mode
+     */
+    public static getEditParam() : string | null {
+        return new URL(PathUtils.getCurrentURL()).searchParams.get(Constants.AEM_MODE_KEY);
+    }
+
+    /**
+     * Generates clientlib url
+     * @returns {string[]} all the clientlib urls
+     */
+    public static generateClientLibsUrl() : string[] {
+        const clientlibs: string[] =  Array<string>();
+        const path = '/etc.clientlibs/cq/gui/components/authoring/editors/clientlibs/internal/';
```

Comment:
I am wondering about extracting this and all such weird hardcoded constants to some AEM-Constants file what do you think ?

Reply:
I had exactly the same thought! Then I think we should do it